### PR TITLE
replace global vault handlers with newVaultHandlers()

### PR DESCRIPTION
**Pull request by thyton**
### Description
This PR addresses the data race caused by #27394 after merged into the ENT repo
```
 =============== test-go (4) ===========================
WARNING: DATA RACE
Write at 0x00c002d54180 by goroutine 63655:
  runtime.mapassign_faststr()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/runtime/map_faststr.go:203 +0x0
  maps.Copy[go.shape.map[string]github.com/hashicorp/vault/sdk/physical.Factory,go.shape.map[string]github.com/hashicorp/vault/sdk/physical.Factory,go.shape.string,go.shape.func(map[string]string, github.com/hashicorp/go-hclog.Logger) (github.com/hashicorp/vault/sdk/physical.Backend, error)]()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/maps/maps.go:55 +0x167
  github.com/hashicorp/vault/command.extendAddonCommands()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/commands_full.go:94 +0x36
  github.com/hashicorp/vault/command.initCommands()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/commands.go:175 +0x56
  github.com/hashicorp/vault/command.RunCustom()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/main.go:225 +0xfee
  github.com/hashicorp/vault/command.execPKIVerifyNonJson()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_verify_sign_test.go:150 +0x2d6
  github.com/hashicorp/vault/command.execPKIVerifyJson()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_verify_sign_test.go:131 +0x92
  github.com/hashicorp/vault/command.runPkiListIntermediateTests()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_list_intermediate_test.go:211 +0x212a
  github.com/hashicorp/vault/command.TestPKIListIntermediate()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_list_intermediate_test.go:36 +0x7a
  testing.tRunner()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/testing/testing.go:1689 +0x21e
  testing.(*T).Run.gowrap1()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/testing/testing.go:1742 +0x44

Previous write at 0x00c002d54180 by goroutine 63654:
  runtime.mapassign_faststr()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/runtime/map_faststr.go:203 +0x0
  maps.Copy[go.shape.map[string]github.com/hashicorp/vault/sdk/physical.Factory,go.shape.map[string]github.com/hashicorp/vault/sdk/physical.Factory,go.shape.string,go.shape.func(map[string]string, github.com/hashicorp/go-hclog.Logger) (github.com/hashicorp/vault/sdk/physical.Backend, error)]()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/maps/maps.go:55 +0x167
  github.com/hashicorp/vault/command.extendAddonCommands()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/commands_full.go:94 +0x36
  github.com/hashicorp/vault/command.initCommands()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/commands.go:175 +0x56
  github.com/hashicorp/vault/command.RunCustom()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/main.go:225 +0xfee
  github.com/hashicorp/vault/command.createComplicatedIssuerSetUpWithIssueIntermediate()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_issue_intermediate_test.go:151 +0x1653
  github.com/hashicorp/vault/command.TestPKIIssueIntermediate()
      /home/runner/actions-runner/_work/vault-enterprise/vault-enterprise/command/pki_issue_intermediate_test.go:35 +0x6b
  testing.tRunner()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/testing/testing.go:1689 +0x21e
  testing.(*T).Run.gowrap1()
      /home/runner/actions-runner/_work/_tool/go/1.22.4/x64/src/testing/testing.go:1742 +0x44
 ```

The PR replaces all global vault handlers in `commands.go` with `newVaultHandlers()` that returns `*vaultHandlers struct` that encapsulates all vault handlers. https://github.com/hashicorp/vault-enterprise/pull/6084's been created to test and address the issue in the ENT repo.

### CODE REVIEW

File: command/commands.go [hunk 1]
```diff
@@ -129,50 +129,106 @@ const (
...
+//		// This is also available in the plugin catalog, but is here due to the need to
+//		// automatically mount it.
+//		"kv": logicalKv.Factory,
+//	}
+//
+//	serviceRegistrations = map[string]sr.Factory{
+//		"consul":     csr.NewServiceRegistration,
+//		"kubernetes": ksr.NewServiceRegistration,
+//	}
+//)
```
Comment (peteski22) [inline command/commands.go:228]:
```suggestion
```

❓ is this to be removed?

File: command/commands.go [hunk 2]
```diff
@@ -129,50 +129,106 @@ const (
...
 
-var (
-	physicalBackends = map[string]physical.Factory{
-		"inmem_ha":               physInmem.NewInmemHA,
-		"inmem_transactional_ha": physInmem.NewTransactionalInmemHA,
-		"inmem_transactional":    physInmem.NewTransactionalInmem,
-		"inmem":                  physInmem.NewInmem,
-		"raft":                   physRaft.NewRaftBackend,
-	}
+type vaultHandlers struct {
```
Comment (peteski22) [inline command/commands.go:132]: Could this have a go doc please?

File: command/commands.go [hunk 3]
```diff
@@ -129,50 +129,106 @@ const (
...
+	serviceRegistrations map[string]sr.Factory
+}
 
-	loginHandlers = map[string]LoginHandler{
-		"cert":  &credCert.CLIHandler{},
-		"oidc":  &credOIDC.CLIHandler{},
-		"token": &credToken.CLIHandler{},
-		"userpass": &credUserpass.CLIHandler{
-			DefaultMount: "userpass",
+func newMinimalVaultHandlers() *vaultHandlers {
```
Comment (peteski22) [inline command/commands.go:141]: Could this have a go doc please?

File: command/commands.go [hunk 4]
```diff
@@ -129,50 +129,106 @@ const (
...
-	auditBackends = map[string]audit.Factory{
-		"file":   audit.NewFileBackend,
-		"socket": audit.NewSocketBackend,
-		"syslog": audit.NewSyslogBackend,
-	}
-
-	credentialBackends = map[string]logical.Factory{
-		"plugin": plugin.Factory,
-	}
+func newVaultHandlers() *vaultHandlers {
```
Comment (peteski22) [inline command/commands.go:180]: Can this have a go doc please?

File: command/command_test.go [hunk 5]
```diff
@@ -122,11 +122,11 @@ func testVaultServerWithKVVersion(tb testing.TB, kvVersion string) (*api.Client,
 func testVaultServerAllBackends(tb testing.TB) (*api.Client, func()) {
 	tb.Helper()
 
+	handlers := newVaultHandlers()
 	client, _, closer := testVaultServerCoreConfig(tb, &vault.CoreConfig{
-		CredentialBackends: credentialBackends,
-		AuditBackends:      auditBackends,
-		LogicalBackends:    logicalBackends,
-		BuiltinRegistry:    builtinplugins.Registry,
```
Comment (stevendpclark) [inline command/command_test.go:129]: Was dropping initializing BuiltinRegistry on purpose here?

File: command/commands.go (hunk 2 above)
Comment (thyton) [reply command/commands.go:133]: Added in [c293ac0](https://github.com/hashicorp/vault/pull/27515/commits/c293ac0eb382666c16170377890ab8d37e8311af).

File: command/commands.go (hunk 3 above)
Comment (thyton) [reply command/commands.go:143]: Added in [c293ac0](https://github.com/hashicorp/vault/pull/27515/commits/c293ac0eb382666c16170377890ab8d37e8311af).

File: command/commands.go (hunk 4 above)
Comment (thyton) [reply command/commands.go:184]: Added in [c293ac0](https://github.com/hashicorp/vault/pull/27515/commits/c293ac0eb382666c16170377890ab8d37e8311af).

File: command/command_test.go (hunk 5 above)
Comment (thyton) [reply command/command_test.go:129]: Oops. Reverted the drop in [c293ac0](https://github.com/hashicorp/vault/pull/27515/commits/c293ac0eb382666c16170377890ab8d37e8311af)

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**